addons/mail/models/mail_channel_partner.py

```diff
@@ -50,6 +51,19 @@ def init(self):
         ("partner_or_guest_exists", "CHECK((partner_id IS NOT NULL AND guest_id IS NULL) OR (partner_id IS NULL AND guest_id IS NOT NULL))", "A channel member must be a partner or a guest."),
     ]
 
+    @api.depends('channel_id.message_ids', 'seen_message_id')
+    def _compute_message_unread_counter(self):
+        unread_domain = [
+            ('model', '=', 'mail.channel'),
+            ('res_id', '=', self.channel_id.id),
+            ('message_type', 'not in', ('notification', 'user_notification')),
+        ]
+        if self.seen_message_id:
+            unread_domain.append(('id', '>', self.seen_message_id.id))
```

Comment:
`self.seen_message_id` is still not batch compatible, I wonder if `read_group` is even possible in this case, as opposed to making the query manually.
For each channel, it should take the `seen_message_id` of the corresponding current channel_partner.